dts/bindings/arm/nordic,nrf-spu.yaml

```diff
@@ -10,3 +10,13 @@ properties:
 
     interrupts:
       required: true
+
+    ram-regions:
+      type: array
+      description: number of lockable ram regions and granularity
```

Comment:
Maybe specify (kB) in granularity? Same for flash regions.
And I 'd actually rename to "region size" instead of "granularity"

Reply:
Should we be specific or should we annotate the device three with kB ?

I'm fine with saying region size instead of granularity.

Reply:
I think it is better to be specific here in the binding file, where we describe what the entries are.
@anangl feel free to comment.

Yes, region size is exactly what we need.
Nit: in the description replace "ram" -> "RAM" (suggestion)

Reply:
> I think it is better to be specific here in the binding file, where we describe what the entries are.

I agree, the binding description is the place where it should be specified what a given property is supposed to contain.

Reply:
Changed it accordingly to your comments I think.

Reply:
One question is if we want to have one or two properties?
`flash/ram-region` and `flash/ram-region-block-size`/`flash/ram-region-size`

Or is it better to group them into an array property?

Reply:
@anangl ^^

Reply:
@galak Any reasoning other than a feeling?

I can see the argument that it would be more readable when calling the device three configurations, however, the values are actually coupled so they should be grouped.

They are to be used in an out-of-tree code that needs some information about what the granularity of the IDAU is.

Reply:
> One question is if we want to have one or two properties?

I don't have a strong opinion about this. To me those arrays are fine, as it's good to have these values paired this way.

> Also if they should be annotated in hexadecimal for size instead of decimals

Let's stay with decimal values. That's what is used in Product Specification.